src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,81 @@
+package duke;
+
+import task.Deadline;
+import task.Event;
+import task.Task;
+import task.Todo;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Loads, parses and saves data from/to specified file destination.
+ */
+public class Storage {
+    private String filepath;
+
+    /**
+     * Creates Storage object with specified file destination.
+     * @param filepath File destination.
+     */
+    Storage(String filepath) {
+        this.filepath = filepath;
+    }
+
+    /**
+     * Loads the tasks from existing list in the file.
+     * @return List of existing tasks.
+     * @throws FileNotFoundException Specified file is not found.
+     * @throws ParseException Invalid variable to parse.
+     * @throws DukeException Invalid actions / missing values.
+     */
+    ArrayList<Task> load() throws FileNotFoundException, ParseException, DukeException {
+        // pass the path to the file as a parameter
+        File file = new File(filepath);
+        Scanner s = new Scanner(file);
+
+        SimpleDateFormat formatter = new SimpleDateFormat("dd/MM/yyyy HH:mm");
+
+        ArrayList<Task> tasks = new ArrayList<>();
+
+        while (s.hasNextLine()) {
+            String[] task = s.nextLine().split(" \\| ");
+            Task taskToAdd;
+
+            switch (task[0]) {
+            case "D":
+                taskToAdd = new Deadline(task[2], task[1].equals("+"), formatter.parse(task[3]));
+                break;
+            case "E":
+                taskToAdd = new Event(task[2], task[1].equals("+"), formatter.parse(task[3]));
+                break;
+            case "T":
+                taskToAdd = new Todo(task[2], task[1].equals("+"));
+                break;
+            default:
+                throw new DukeException();
+            }
+            tasks.add(taskToAdd);
+        }
+        return tasks;
```

Comment:
Might be good to close the Scanner after you're done!
```suggestion
        s.close();
        return tasks;
```